dask_expr/io/parquet.py

```diff
@@ -542,25 +542,44 @@ def _dataset_info(self):
         meta = self.engine._create_dd_meta(dataset_info)
         index = dataset_info["index"]
         index = [index] if isinstance(index, str) else index
-        meta, index, columns = set_index_columns(
-            meta, index, self.operand("columns"), auto_index_allowed
+        meta, index, all_columns = set_index_columns(
+            meta, index, None, auto_index_allowed
         )
         if meta.index.name == NONE_LABEL:
             meta.index.name = None
-        dataset_info["meta"] = meta
+        dataset_info["base_meta"] = meta
         dataset_info["index"] = index
-        dataset_info["columns"] = columns
+        dataset_info["all_columns"] = all_columns
 
         return dataset_info
 
     @property
     def _meta(self):
-        meta = self._dataset_info["meta"]
+        meta = self._dataset_info["base_meta"]
+        columns = _convert_to_list(self.operand("columns"))
         if self._series:
-            column = _convert_to_list(self.operand("columns"))[0]
-            return meta[column]
+            assert columns
```

Comment:
I don't like this pattern very much. could we do ``assert len(columns) > 0``?

Reply:
Yeah, sounds good.